### ISSUE TITLE
SonarQube line comments are not displayed in diff view

### ISSUE THREAD

**Issue by johnou** [gabrie-allaigre/sonar-gitlab-plugin#15]
When viewing a merge request in diff view the SonarQube comments are not displayed, is there maybe a bug with how the comment is added or a different API endpoint / field for having the comment in the diff view? cc. @gabrie-allaigre

**Comment by johnou** [comment]
@jairbubbles do you see the same behaviour?

**Comment by jairbubbles** [comment]
@johnou No but I do not have the same exact version. I use a version from a fork which uses the new GitLab API. Maybe it's linked ? (I don't remember the name of the fork, I'm not at work so I can't tell you which one) 
Which GitLab version do you use ? For me it's 8.17.

**Comment by johnou** [comment]
We are currently on GitLab 8.16.5-ee, SonarQube 6.2. I will look through the forks soon, could I trouble you for the jar?

**Comment by jairbubbles** [comment]
@johnou I'll give you all the info tomorrow when I'm at work!

**Comment by gabrie-allaigre** [comment]
Hi,
It not same comment in side by side merge request.
https://gitlab.com/gitlab-org/gitlab-ce/issues/14850

**Comment by johnou** [comment]
ow, nice find.

**Comment by jairbubbles** [comment]
@johnou  I forked this repo https://github.com/Jojo255/sonar-gitlab-plugin

**Comment by johnou** [comment]
@jairbubbles haha they made the same mistake as me, the GitLab API plugin they swapped to isn't the official 😆 can you double check that comments come up in the diff? According to the issue @gabrie-allaigre linked to it should not be possible since the API doesn't exist..

**Comment by jairbubbles** [comment]
@johnou  I had a screenshot to prove you it work but I lost it and now I can't see the comment anymore. New commits were made on the MR so that might explain why it's not there anymore.

**Comment by jairbubbles** [comment]
@johnou If I go back to a previous version I get the message:
![image](https://cloud.githubusercontent.com/assets/15875066/23549329/00ba3a0c-000c-11e7-9008-d153d1dff925.png)

@gabrie-allaigre https://github.com/kakawait/sonar-gitlab-plugin seems quite active should be great to merge with both forks isn't it ? Why not put it on https://github.com/SonarQubeCommunity ? It seems to be the place for such plugins.

**Comment by johnou** [comment]
@jairbubbles see https://github.com/gabrie-allaigre/sonar-gitlab-plugin/pull/13#issuecomment-283326111

**Comment by jairbubbles** [comment]
@johnou  Not sure I follow you.
https://github.com/SonarSource != https://github.com/SonarQubeCommunity

**Comment by johnou** [comment]
@jairbubbles yep but that org is for the update center Simon mentions, right?

**Comment by jairbubbles** [comment]
@johnou It seems so yes. Having a central depot in SonarqubeCommunity would be a first step I guess to make it an official plugin.

**Comment by gabrie-allaigre** [comment]
@jairbubbles My goal is to have the plugin on sonar directly. But I add functionnaly (multi hash) and test for code cover.
https://sonarqube.com/dashboard?id=com.talanlabs%3Asonar-gitlab-plugin

Also my other plugin https://github.com/gabrie-allaigre/sonar-auth-gitlab-plugin

**Comment by jairbubbles** [comment]
@gabrie-allaigre That's even better! Thx anyway for making the plugin in the first place.

**Comment by johnou** [comment]
Multi hash?

**Comment by johnou** [comment]
@gabrie-allaigre multi hash?

**Comment by gabrie-allaigre** [comment]
@johnou In Git, when you push, there are several commit. Currently I have only the last commit, the multi hash is to take all the commits of the branch and to comment inline.

**Comment by johnou** [comment]
@gabrie-allaigre again, little off topic but do you know if a lot of people use old version of Sonar? I recall seeing a comment saying there was a file cache hack which could be removed after updating to latest Sonar API. Maybe after implementing multi-hash you could create a branch then bump the major version and only support latest Sonar wdyt?

**Comment by gabrie-allaigre** [comment]
@johnou File cache hack is already deleting in new version 2.0.0. It's for sonarqube >= 5.6.
For older sonar versions, use version 1.7.0

**Comment by kakawait** [comment]
I don't want to be a competitor, but I've implemented multi-hashed support on a fork (https://github.com/kakawait/sonar-gitlab-plugin).

I don't think I can create a MR because code base is now completely different (rewritten for JDK8 + official Gitlab lib) but feel free to look at if you want

**Comment by gabrie-allaigre** [comment]
@kakawait, Yes thanks, I add functionnality.I will change for GitLabAPI when it will have integrated my MR on the proxy support.

**Comment by kakawait** [comment]
Maybe off-topic too but _GitLab_ also have a issue about commenting MR, see open issue https://gitlab.com/gitlab-org/gitlab-ce/issues/26606

Thus on our current _GitLab_ version that code does not work as expected https://github.com/gabrie-allaigre/sonar-gitlab-plugin/blob/master/src/main/java/com/talanlabs/sonar/plugins/gitlab/CommitFacade.java#L94 (due to _GitLab_ bug is not bug from plugin itself)

**Comment by johnou** [comment]
@gabrie-allaigre still working on multi-hash or is it already in master?

**Comment by gabrie-allaigre** [comment]
@johnou yes work in progress on multi-hash.
Multi-hash will be just one more option

**Comment by gabrie-allaigre** [comment]
@johnou multi-hash is done on master

**Comment by johnou** [comment]
@gabrie-allaigre sweet, I'll test it out on Monday!

**Comment by johnou** [comment]
@gabrie-allaigre I tried out multi-hash with a feature branch and multiple commits, what I found is that if the first commit contains a blocker issue which has been resolved in the following commit it may still result in the build to failing.

**Comment by johnou** [comment]
@kakawait did you run into the same issue with your branch / multi-hash implementation?

**Comment by gabrie-allaigre** [comment]
@johnou SonarQube analyse last commit only

**Comment by gabrie-allaigre** [comment]
@johnou Do you still have the problem?

**Comment by kakawait** [comment]
@johnou I think I don't have such issue because I always perform sonar analysis on latest commit so if issue has been fixed sonar shouldn't raise issue. I only use multi hashes to find the proper line to comment on github

**Comment by johnou** [comment]
@gabrie-allaigre yes, I ended up reverting for now. Unfortunately it doesn't work too well with our development flow.

**Comment by gabrie-allaigre** [comment]
@johnou Can you tell me what's wrong?
How is your development flow ?
Personally we do not use multi-hash, it has no interest in our flow

**Comment by johnou** [comment]
@gabrie-allaigre I _suspect_ when working on a feature branch and changes need to be merged from master into the feature branch those changes (which are not part of the feature branch) are included in the analysis.

**Comment by gabrie-allaigre** [comment]
@johnou It is strange, they would need an example of log of the feature-branch and the master

**Comment by johnou** [comment]
New Gitlab issue for tracking the feature request is https://gitlab.com/gitlab-org/gitlab-ce/issues/20901

**Comment by kamihouse** [comment]
Hi guys.  
I do not know how to enable the function **Executing post-job Gitlab Commit Issue Publisher** to send comments of issue reports to gitlab.

I'm running some tests on a PHP project. Am I forgetting to inform you of any configuration?

File: `sonar-scanner.properties`
```
sonar.host.url=$SONAR_URL
sonar.projectKey=xyz
sonar.projectName=xyz
sonar.projectVersion=1.0.0
sonar.analysis.mode=preview
sonar.sourceEncoding=UTF-8
```

---

### Context:

* Sonar GitLab Plugin version: 2.0.1
* Gitlab version: 9.3.2
* Jenkins version: 2.60.1

**Comment by gabrie-allaigre** [comment]
Hi,
Add sonar.gitlab.commit_sha & sonar.gitlab.project_id

**Comment by johnou** [comment]
New Gitlab issue for tracking the feature request is https://gitlab.com/gitlab-org/gitlab-ce/issues/20901

**Comment by johnou** [comment]
@jakub-bochenski yes, that's correct.

**Comment by btrajkovski** [comment]
@gabrie-allaigre can we just replace the current comment API for every commit with this new discussion API so instead of leaving comments it leaves the discussion. Is this the only line we need to change https://gitlab.talanlabs.com/gabriel-allaigre/java-gitlab-api/blob/master/src/main/java/com/talanlabs/gitlab/api/v3/services/GitLabAPICommits.java#L134 
to use the new discussions API ?
I would be willing to make this PR if you are fine with this change.
